### PR TITLE
Add a warning when the transaction isolation level is not READ_COMMITED

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref #24882 

@PVince81 @DeepDiver1975

### CODE REVIEW

File: settings/admin.php [hunk 1]
```diff
@@ -103,6 +103,15 @@
 $template->assign('encryptionReady', \OC::$server->getEncryptionManager()->isReady());
 $template->assign('externalBackendsEnabled', $externalBackends);
 
+/** @var \Doctrine\DBAL\Connection $connection */
+$connection = \OC::$server->getDatabaseConnection();
+try {
+	$template->assign('invalidTransactionIsolationLevel', $connection->getTransactionIsolation() !== \Doctrine\DBAL\Connection::TRANSACTION_READ_COMMITTED);
```
Comment (DeepDiver1975) [inline settings/admin.php]: Is this required for all dbms or only mysql?

**Comment by DeepDiver1975** [comment]
We should backport this to all stable releases

File: settings/admin.php (hunk 1 above)
Comment (nickvergessen) [inline settings/admin.php]: It is required for all that have a transaction isolation level:
Postgres: https://www.postgresql.org/docs/current/static/transaction-iso.html
Oracle: http://docs.oracle.com/cd/E11882_01/server.112/e41084/statements_10005.htm#SQLRF01705

Can't find exact docs for sqlite, it seems to have a boolean for "read uncommited" instead.
So I guess I will disable it for sqlite

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by rullzer** [comment]
:+1:

**Comment by PVince81** [comment]
@nickvergessen please submit the backport PRs, thanks

CC @cmonteroluque @DeepDiver1975

**Comment by nickvergessen** [comment]
9.0: https://github.com/owncloud/core/issues/24915
8.2: https://github.com/owncloud/core/issues/24916
8.1: https://github.com/owncloud/core/issues/24917
8.0: https://github.com/owncloud/core/issues/24918

**Comment by cmonteroluque** [comment]
thanks @nickvergessen

**Comment by RealRancor** [comment]
Ref to the doc part: https://github.com/owncloud/documentation/issues/2477

There definitely will be questions about that coming up

**Pull request merged**